lia-ui/src/main/java/com/lithium/community/android/ui/components/activities/LiSupportHomeActivity.java

```diff
@@ -158,13 +181,95 @@ public void onClick(View view) {
+        Menu menu = nv.getMenu();
+        for (int i = 0; i < menu.size(); i++) {
+            MenuItem item = menu.getItem(i);
+            int id = item.getItemId();
+            boolean visibility = (isUserLoggedIn && (id != R.id.li_main_navigation_menu_login)
+                    || (!isUserLoggedIn && (id == R.id.li_main_navigation_menu_login)));
+            item.setVisible(visibility);
+        }
+        View headerView = nv.getHeaderView(0);
+        final ImageView icon = headerView.findViewById(R.id.li_main_navigation_header_icon);
+        final TextView title = headerView.findViewById(R.id.li_main_navigation_header_title);
+        final TextView subTitle = headerView.findViewById(R.id.li_main_navigation_header_sub_title);
+
+        subTitle.setText(LiSDKManager.getInstance().getCredentials().getCommunityUri().toString());
+
+        if (isUserLoggedIn) {
+            title.setText(String.format("%1$s\n<%2$s>", user.getLogin(), user.getEmail()));
+            String url = user.getAvatar() != null ? user.getAvatar().getProfile() : null;
+            if (TextUtils.isEmpty(url)) {
+                return;
+            }
+            Target target = new Target() {
+                @Override
+                public void onBitmapLoaded(Bitmap bitmap, Picasso.LoadedFrom from) {
+                    icon.setImageBitmap(bitmap);
+                }
+
+                @Override
+                public void onBitmapFailed(Drawable errorDrawable) {
+                    icon.setImageDrawable(errorDrawable);
+                }
+
+                @Override
+                public void onPrepareLoad(Drawable placeHolderDrawable) {
+
+                }
+            };
+            icon.setTag(target);
+            Picasso.with(this).load(url).resize(80, 80)
+                    .onlyScaleDown().into(target);
+        } else {
+            icon.setImageResource(R.mipmap.ic_launcher);
+            title.setText(R.string.li_main_navigation_header_title);
+        }
+    }
+
+    @Override
+    public Intent registerReceiver(BroadcastReceiver receiver, IntentFilter filter) {
+        isLoginReceiverRegistered = true;
+        return super.registerReceiver(receiver, filter);
+    }
+
+    @Override
+    public void unregisterReceiver(BroadcastReceiver receiver) {
+        isLoginReceiverRegistered = false;
+        super.unregisterReceiver(receiver);
+    }
+
+    private class NavigationListener extends ActionBarDrawerToggle {
+
+        public NavigationListener(Activity context) {
+            super(context, drawerLayout, R.string.li_main_navigation_drawer_open, R.string.li_main_navigation_drawer_close);
+        }
     }
 
     @Override
     protected void onDestroy() {
         super.onDestroy();
         unregisterReceiver(questionsTabUnreadReplyReceiver);
         unregisterReceiver(createMessageSuccessReceiver);
+        if (isLoginReceiverRegistered) {
+            unregisterReceiver(receiver);
```

Comment:
a call to onDestroy() is not always guaranteed.. Can you explain the lifecycle of this receiver?

Reply:
Receiver here does nothing but a UI Action callback, it's the current design.
Lifecycle of receiver is not in line with that of activity and it need not be.
Receiver is registered once login is initiated, and it is unregistered when login succeeds or fails.
But if login was cancelled, the registry stays put, Hence this code is the extra check to not leak the receiver.